tests/test_time.py

```diff
@@ -376,3 +376,232 @@ def test_naturaltime_minimum_unit_explicit(minimum_unit, seconds, expected):
+        (dt.timedelta(days=120), "months", "%0.2f", "3.93 months"),
+        (dt.timedelta(days=183), "years", "%0.1f", "0.5 years"),
+    ],
+)
+def test_precisedelta_custom_format(val, min_unit, fmt, expected):
+    assert humanize.precisedelta(val, minimum_unit=min_unit, format=fmt) == expected
+
+
+@pytest.mark.parametrize(
+    "val, min_unit, suppress, expected",
+    [
+        (
+            dt.timedelta(microseconds=1200),
+            "microseconds",
+            [],
+            "1 millisecond and 200 microseconds",
+        ),
+        (
+            dt.timedelta(microseconds=1200),
+            "microseconds",
+            ["milliseconds"],
+            "1200 microseconds",
+        ),
+        (
+            dt.timedelta(microseconds=1200),
+            "microseconds",
+            ["microseconds"],
+            "1.20 milliseconds",
+        ),
+        (
+            dt.timedelta(seconds=1, microseconds=200),
+            "microseconds",
+            ["seconds"],
+            "1000 milliseconds and 200 microseconds",
+        ),
+        (
+            dt.timedelta(seconds=1, microseconds=200000),
+            "microseconds",
+            ["milliseconds"],
+            "1 second and 200000 microseconds",
+        ),
+        (
+            dt.timedelta(seconds=1, microseconds=200000),
+            "microseconds",
+            ["milliseconds", "microseconds"],
+            "1.20 seconds",
+        ),
+        (
+            dt.timedelta(hours=4, seconds=30, microseconds=200),
+            "microseconds",
+            ["microseconds"],
+            "4 hours, 30 seconds and 0.20 milliseconds",
+        ),
+        (
+            dt.timedelta(hours=4, seconds=30, microseconds=200),
+            "microseconds",
+            ["seconds"],
+            "4 hours, 30000 milliseconds and 200 microseconds",
+        ),
+        (
+            dt.timedelta(hours=4, seconds=30, microseconds=200),
+            "microseconds",
+            ["seconds", "milliseconds"],
+            "4 hours and 30000200 microseconds",
+        ),
+        (
+            dt.timedelta(hours=4, seconds=30, microseconds=200),
+            "microseconds",
+            ["hours"],
+            "240 minutes, 30 seconds and 200 microseconds",
+        ),
+        (
+            dt.timedelta(hours=4, seconds=30, microseconds=200),
+            "microseconds",
+            ["hours", "seconds", "milliseconds", "microseconds"],
+            "240.50 minutes",
+        ),
+    ],
+)
+def test_precisedelta_suppress_units(val, min_unit, suppress, expected):
+    assert (
+        humanize.precisedelta(val, minimum_unit=min_unit, suppress=suppress) == expected
+    )
+
+
+def assertRaises(func, exception_class, *args, **kargs):
+    try:
+        func(*args, **kargs)
+    except Exception as e:
+        if isinstance(e, exception_class):
+            assert True
+            return
+
+    assert False
+
+
```

Comment:
We can use `pytest.raises` instead

https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions

```suggestion
```